pkg/propertychecker/azure/mock_server.go

```diff
@@ -0,0 +1,78 @@
+//
+//Copyright (c) Microsoft Corporation.
+//Licensed under the MIT license.
+
+package azure
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"google.golang.org/protobuf/encoding/protojson"
+
+	computev1 "go.goms.io/fleet/apis/protos/azure/compute/v1"
+	"go.goms.io/fleet/pkg/clients/httputil"
+)
+
+// CreateAttributeBasedVMSizeRecommenderServer creates a mock HTTP server for testing AttributeBasedVMSizeRecommenderClient.
+func CreateAttributeBasedVMSizeRecommenderServer(t *testing.T, httpStatusCode int) *httptest.Server {
```

Comment:
no need to put it in a separate file. It won't conflict with backport changes.

no need to make it public and it will be used internally only?